src/main/database.cpp

```diff
@@ -47,10 +47,10 @@ Database::Database(std::string databasePath, SystemConfig systemConfig)
     : databasePath{std::move(databasePath)}, systemConfig{systemConfig} {
     initLoggers();
     logger = LoggerUtils::getLogger(LoggerConstants::LoggerEnum::DATABASE);
-    initDBDirAndCoreFilesIfNecessary();
     bufferManager = std::make_unique<BufferManager>(this->systemConfig.bufferPoolSize);
     memoryManager = std::make_unique<MemoryManager>(bufferManager.get());
     queryProcessor = std::make_unique<processor::QueryProcessor>(this->systemConfig.maxNumThreads);
+    initDBDirAndCoreFilesIfNecessary();
```

Comment:
Not a required change. I moved it because the initialization of `bufferManager`, `memoryManager`, and `queryProcessor` has nothing to do with the initialization of dir and files.